Container-License-Compliance/flows-annotations.md

```diff
@@ -0,0 +1,137 @@
+
+Ensure you understand what code you imported, from where.
+
+Ensure you understand any license implications of that code.
+
+Ensure you list that code in your BOM if applicable.
+
+Ensure you list that code in your approval request if applicable.
+
+Ensure if you download code into your layers, you use a secure channel to download and/or check the validity of SHA/checksums if available.
+
+
+## (4) Importing code from inside
+
+Ensure you understand if all the imported code is from your project, or imported from other parts of intel as well.
+
+Ensure all of the code appears in your approval request slide deck, as appropriate.
+
+If you are importing code from another project, ensure that project already has an approval, or be prepared to roll that code into your project for approval.
+
+
+## (5) Are you releasing your DockerFile
+
+Are you shipping your DockerFile (for instance, by pushing to GitHub, or supplying to end users via any other method such as email, ftp etc.)?
+
+Ensure your DockerFile containers a license.
+
+If you imported or copied any of your Dockerfile from elsewhere (see (1)), ensure that your (re)license is compatible with all such artifacts.
+
+(NB: this should trigger a process to approve an external delivery)
+
+
+## (6) Publishing your layer to an existing image hub/registry
+
+Are you publishing your layer to an existing public repository, e.g. `docker push` to `hub.docker.com`.
+
+Ensure you understand everything that is in your layers (the layers you authored in your DockerFile).
+
+Minor technical point: `docker push` actually pushes *all* the layers of your image to the registry to prove it has access to them. If the layers already exist on the registry, after verification, the data is discarded. Thus, you are not really publishing the base layers to others.
+
+(NB: this should trigger a process to approve an external delivery)
+
+
+## (7) Publishing your layer to your own image hub/registry (dissuaded)
+
+Similar to (6), but in this case you are publishing all the layers in your image, including all layers obtained from outside sources (such as in (2) and (3)).
+
+Ensure you have the legal rights to publish these layers.
+
+Ensure you meet any legal obligations that come with these layers.
+
+(NB: this should trigger a process to approve an external delivery)
+
+
+## (8) Publishing a (full) docker image (dissuaded)
+
+If you are publishing a full docker image (not just a docker layer) ---for instance via `docker save`, to a tarfile--- then that image file will contain all the layers of your image, including those obtained in (2) and (3). You are therefore publishing and shipping those layers.
+
+Ensure you have the legal rights to publish these layers.
+
+Ensure you meet any legal obligations that come with these layers.
+
+(NB: this should trigger a process to approve an external delivery)
+
+
+## (9) Customer imports your image
```

Comment:
Perhaps is is better to write "a 3rd party" instead of customer. This comment applies to all appearances of the term customer

Reply:
"external recipient"